### ISSUE TITLE
How widely should HTML's 'aspect-ratio' presentational attribute be applied?

### ISSUE THREAD

**Issue by tabatkins** [w3c/csswg-drafts#5608]
In <https://github.com/whatwg/html/pull/6032>, I'm fixing HTML to define that `<img width height>` constitutes an 'aspect-ratio' presentational hint (setting 'aspect-ratio' at the presentation-hint layer, lower specificity than any rule). (Previously, this was defined as width/height attributes actually giving the element an intrinsic aspect ratio.)

The text I'm replacing only defined this mapping for `img` elements, but it seems reasonable that this should apply to all element that currently process 'width' and 'height' attributes as presentational hints: `embed`, `iframe`, `img`, `object`, and `video`. Does this sound reasonable to other people?

**Comment by fantasai** [comment]
No, it should apply to video, but embed/object/iframe often represent elements that have a width and a height but no aspect ratio. So it should not map to those.

I feel like there's existing discussion about this over in the WHATWG space...

**Comment by zcorpan** [comment]
What @fantasai said.

Although `embed` and `object` *can* be an image (and so have an intrinsic aspect ratio), they can also be a nested browsing context (no intrinsic aspect ratio).

There is however `input type=image` which could also do this.

For `canvas`, the spec currently says 

> The intrinsic dimensions of the canvas element when it represents embedded content are equal to the dimensions of the element's bitmap.

https://html.spec.whatwg.org/multipage/canvas.html#the-canvas-element:intrinsic-dimensions

**Comment by fantasai** [comment]
So basically `img`, `input type=image`, `video`, and `canvas` should have an aspect-ratio mapping, but none of the others should.

**Comment by fantasai** [comment]
The edits appear to be in HTML now, so closing.
